xdsl/ir.py

```diff
@@ -317,6 +317,9 @@ def __str__(self) -> str:
 
 _D = TypeVar("_D", bound="Data[Any]")
 
+AttributeTCov = TypeVar("AttributeTCov", bound=Attribute, covariant=True)
+AttributeT = TypeVar("AttributeT", bound=Attribute)
```

Comment:
```suggestion
AttributeTInv = TypeVar("AttributeTInv", bound=Attribute)
```

I would prefer to have this name. Feels more descriptive.

Reply:
It might be for TypeVar, but in terms of naming, it feels to me like AttributeT itself could be everything...

Reply:
Yes, we should define a proper naming scheme for it.
I would personally feel that `AttributeT` is enough, but if you guys want to opt for `AttributeInvT`, I'm completely fine with it as well! It would also make sure people actually think if they should have `Inv`, `Cov`, or `Contr`.

Reply:
Might be useful to remind people further down the line of the type if we added Inv?

Reply:
Thinking about it, `AttributeCovT` is better than `AttributeTCov`. I'd still want to go for `AttributeT` instead of `AttributeInvT`, but we can change that later with a simple find/replace. Let's go for `Attribute{Variance}T` for now, and change later if we feel like it